File: library/src/test/java/com/proxerme/library/api/ProxerResponseCallAdapterFactoryTest.java
package com.proxerme.library.api;

import com.proxerme.library.entitiy.notifications.NewsArticle;
import com.squareup.moshi.Types;
import org.junit.Before;
import org.junit.Test;

import java.lang.annotation.Annotation;
import java.util.List;

import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;

/**
 * @author Ruben Gees
 */
public class ProxerResponseCallAdapterFactoryTest {

    private ProxerResponseCallAdapterFactory factory;

    @Before
    public void setUp() {
        factory = new ProxerResponseCallAdapterFactory();
    }

    @Test
    public void testGet() {
        assertThat(factory.get(Types.newParameterizedType(ProxerCall.class, NewsArticle.class),
                new Annotation[0], null)).isNotNull();
    }

    @Test
    public void testGetInvalidType() {
        assertThat(factory.get(Types.newParameterizedType(List.class, NewsArticle.class),
                new Annotation[0], null)).isNull();
    }

    @Test
    public void testGetNoParameter() {
        assertThat(factory.get(List.class, new Annotation[0], null)).isNull();
    }
}

File: library/src/test/kotlin/me/proxer/library/internal/adapter/ProxerResponseCallAdapterFactoryTest.kt
package me.proxer.library.internal.adapter

import com.squareup.moshi.Types
import me.proxer.library.ProxerCall
import me.proxer.library.entity.notifications.NewsArticle
import org.amshove.kluent.shouldBeNull
import org.amshove.kluent.shouldNotBeNull
import org.junit.jupiter.api.Test
import retrofit2.Retrofit

/**
 * @author Ruben Gees
 */
class ProxerResponseCallAdapterFactoryTest {

    private val factory = ProxerResponseCallAdapterFactory()

    private val retrofit = Retrofit.Builder()
        .baseUrl("https://example.com")
        .build()

    @Test
    fun testGet() {
        val proxerCallType = Types.newParameterizedType(ProxerCall::class.java, NewsArticle::class.java)

        factory.get(proxerCallType, emptyArray(), retrofit).shouldNotBeNull()
    }

    @Test
    fun testGetInvalidType() {
        val listType = Types.newParameterizedType(List::class.java, NewsArticle::class.java)

        factory.get(listType, emptyArray(), retrofit).shouldBeNull()
    }

    @Test
    fun testGetNoParameter() {
        factory.get(List::class.java, emptyArray(), retrofit).shouldBeNull()
    }
}
